jose.py

```diff
@@ -524,12 +525,17 @@ def _validate(claims, validate_claims, expiry_seconds):
         _check_not_before(now, not_before)
 
 
-def _jwe_hash_str(ciphertext, iv, adata='', legacy=False):
+def _jwe_hash_str(ciphertext, iv, adata='', version=_TEMP_VER):
     # http://tools.ietf.org/html/
     # draft-ietf-jose-json-web-algorithms-24#section-5.2.2.1
-    if legacy:
+    # Both tokens without version and with version 1 should be ignored in
+    # the future as they use incorrect hashing. The version parameter
+    # should also be removed.
+    if not version:
```

Comment:
`0` here is equivalent to `None` again. Both have the same meaning for this library, which is: **there is no recognizable version number**. Hence the simple condition.